### ISSUE TITLE
Allow factional numbers in the text control for size tests.

### ISSUE THREAD

**Issue by owenca** [HaikuArchives/Filer#74]
The decimal mark should be disallowed for the "bytes" unit and localized to either '.' or ',' depending on the current locale.

Also, dragging & dropping a file should set the text to a value and unit that makes the most sense, e.g., 1.23 MiB instead of 1259.52 KiB or even worse, 1289748 bytes.

**Comment by owenca** [comment]
It seems Haiku doesn't support locale-sensitive decimal mark.

**Comment by humdingerb** [comment]
Until we can find a solution, why not allow "." and "," in the size text control, and when saving the rule, replace the "," with a "." before converting to a float if that isn't done automatically?

When loading the rule, you can do something like the following to localize the rule's size float:
```
float fromRule = 123.45;
BString result;
BNumberFormat numberFormat;
status_t status = numberFormat.Format(result, fromRule);
if (status == B_OK)
	fPositiveNumberExampleView->SetText(result);
```

**Comment by owenca** [comment]
I need to change the format of the size value saved. It's saved as a string but should be changed to a number.

**Comment by owenca** [comment]
Pulkomandy is implementing `BNumberFormat::Parse()`, which is the reverse of `BNumberFormat::Format()`. Then I can use it to hopefully complete the fix for this issue.

**Comment by owenca** [comment]
Well, It turns out that I don't have to use `BNumberFormat::Parse()` to complete the fix.

**Comment by humdingerb** [comment]
I think it works well enough. Just for completeness, I mention what could be improved (probably once Parse() exists): People currently cannot use 1000-delimiters, e.g. 1,234.56 (or German: 1.234,56). It's not crucial, but what separates Filer from perfection. :)

Let's mark this fixed then with #76 . Thanks!
